src/sort_filter/KalmanTracker.cpp

```diff
@@ -0,0 +1,117 @@
+///////////////////////////////////////////////////////////////////////////////
+// KalmanTracker.cpp: KalmanTracker Class Implementation Declaration
+
+#include "KalmanTracker.h"
+#include <ctime>
+
+using namespace std;
+using namespace cv;
+
+int KalmanTracker::kf_count = 0;
+
+// initialize Kalman filter
+void KalmanTracker::init_kf(
+	StateType stateMat)
+{
+	int stateNum = 7;
+	int measureNum = 4;
+	kf = KalmanFilter(stateNum, measureNum, 0);
+
+	measurement = Mat::zeros(measureNum, 1, CV_32F);
+
+	kf.transitionMatrix = (Mat_<float>(stateNum, stateNum) << 1, 0, 0, 0, 1, 0, 0,
+						   0, 1, 0, 0, 0, 1, 0,
+						   0, 0, 1, 0, 0, 0, 1,
+						   0, 0, 0, 1, 0, 0, 0,
+						   0, 0, 0, 0, 1, 0, 0,
+						   0, 0, 0, 0, 0, 1, 0,
+						   0, 0, 0, 0, 0, 0, 1);

```

Comment:
Other alternative:

```suggestion
	kf.transitionMatrix = (
		Mat_<float>(stateNum, stateNum)
		<< 1, 0, 0, 0, 1, 0, 0,
		   0, 1, 0, 0, 0, 1, 0,
		   0, 0, 1, 0, 0, 0, 1,
		   0, 0, 0, 1, 0, 0, 0,
		   0, 0, 0, 0, 1, 0, 0,
		   0, 0, 0, 0, 0, 1, 0,
		   0, 0, 0, 0, 0, 0, 1
	);
```

...I don't have any strong opinions on which one is more readable.